byzcoin/contracts/contract_deferred_test.go

```diff
@@ -2276,3 +2277,666 @@ func TestDeferred_ScenarioUpdateConfig(t *testing.T) {
+					},
+				},
+			},
+			SignerCounter: []uint64{2},
+		}},
+	}
+	require.Nil(t, ctx.FillSignersAndSignWith(signer))
+
+	_, err = cl.AddTransaction(ctx)
+	require.Nil(t, err)
+
+	myID := ctx.Instructions[0].DeriveID("")
+	pr, err = cl.WaitProof(byzcoin.NewInstanceID(myID.Slice()), 2*genesisMsg.BlockInterval, nil)
+	require.Nil(t, err)
+	require.True(t, pr.InclusionProof.Match(myID.Slice()))
+
+	dataBuf, _, _, err := pr.Get(myID.Slice())
+	require.Nil(t, err)
+	result := byzcoin.DeferredData{}
+	err = protobuf.Decode(dataBuf, &result)
+	require.Nil(t, err)
+
+	require.Equal(t, result.ProposedTransaction, proposedTransaction)
+	require.Equal(t, len(result.ProposedTransaction.Instructions), 1)
+	require.Empty(t, result.ProposedTransaction.Instructions[0].SignerIdentities)
+	require.Empty(t, result.ProposedTransaction.Instructions[0].Signatures)
+
+	local.WaitDone(genesisMsg.BlockInterval)
+
+	rootHash := result.InstructionHashes
+
+	// ------------------------------------------------------------------------
+	// 3. Invoke a first "addProof" with the first signer
+	// ------------------------------------------------------------------------
+
+	identity := signer.Identity()
+	identityBuf, err := protobuf.Encode(&identity)
+	require.Nil(t, err)
+
+	signature, err := signer.Sign(rootHash[0]) // == index
+	require.Nil(t, err)
+	// signature[1] = 0xf
+
+	index := uint32(0)
+	indexBuf := make([]byte, 4)
+	binary.LittleEndian.PutUint32(indexBuf, uint32(index))
+
+	ctx = byzcoin.ClientTransaction{
+		Instructions: []byzcoin.Instruction{{
+			InstanceID: myID,
+			Invoke: &byzcoin.Invoke{
+				ContractID: byzcoin.ContractDeferredID,
+				Command:    "addProof",
+				Args: []byzcoin.Argument{
+					{
+						Name:  "identity",
+						Value: identityBuf,
+					},
+					{
+						Name:  "signature",
+						Value: signature,
+					},
+					{
+						Name:  "index",
+						Value: indexBuf,
+					},
+				},
+			},
+			SignerCounter: []uint64{3},
+		}},
+	}
+	require.Nil(t, ctx.FillSignersAndSignWith(signer))
+
+	_, err = cl.AddTransaction(ctx)
+	require.Nil(t, err)
+
+	proposedTransaction.Instructions[0].SignerIdentities = append(proposedTransaction.Instructions[0].SignerIdentities, identity)
+	proposedTransaction.Instructions[0].Signatures = append(proposedTransaction.Instructions[0].Signatures, signature)
+	result.ProposedTransaction = proposedTransaction
+
+	resultBuf, err := protobuf.Encode(&result)
+	require.Nil(t, err)
+
+	pr, err = cl.WaitProof(byzcoin.NewInstanceID(myID.Slice()), 2*genesisMsg.BlockInterval, resultBuf)
+	require.Nil(t, err)
+	require.True(t, pr.InclusionProof.Match(myID.Slice()))
+
+	dataBuf, _, _, err = pr.Get(myID.Slice())
+	require.Nil(t, err)
+
+	result = byzcoin.DeferredData{}
+	err = protobuf.Decode(dataBuf, &result)
+	require.Nil(t, err)
+
+	// We can not do this test because the identities have to be compared using
+	// the Equal() method.
+	//require.Equal(t, result.ProposedTransaction, proposedTransaction)
```

Comment:
this won't work because `proposedTransaction` doesn't have the signer identities? you could add the expected signer identities to `proposedTransaction` and then compare the hash of `proposedTransaction` with the hash of `result.ProposedTransaction`, then u won't need to do any of the comparison below because the signer identity is a part of the hash.

Reply:
True! Although we can only compute the hash of an instruction (right?)

Reply:
there should be a function that computes the hash for a list of instructions